dietpi/dietpi-config

```diff
@@ -822,160 +822,441 @@ A long (or insufficiently manufactured) cable may required a higher boost settin
 
 			fi
 
-		# RPi
-		elif (( $G_HW_MODEL < 10 )); then
+			# RPi
+			elif (( $G_HW_MODEL < 10 )); then
+
+				local framebuffer_x=$(grep -m1 '^[[:blank:]]*framebuffer_width=' /DietPi/config.txt || vcgencmd get_config framebuffer_width)
+				framebuffer_x=${framebuffer_x#*=}; framebuffer_x=${framebuffer_x:-0}
+				local framebuffer_y=$(grep -m1 '^[[:blank:]]*framebuffer_height=' /DietPi/config.txt || vcgencmd get_config framebuffer_height)
+				framebuffer_y=${framebuffer_y#*=}; framebuffer_y=${framebuffer_y:-0}
+				local current_value=$(grep -m1 '^[[:blank:]]*dtoverlay=vc4-' /DietPi/config.txt | sed 's/^[^=]*=//') #OpenGL check 1st
+
+				#If framebuffer is not set by user then check if hdmi default is detected and being used
+				if [[ "$framebuffer_y" == 0 ]] && [[ "$framebuffer_x" == 0 ]]; then
```

Comment:
bash magic:
- left sided variables never need to be quoted, multiple conditionals can be merged into one statement:
`[[ $framebuffer_y == 0 && $framebuffer_x == 0 ]]`
or as arithmetic check
`(( $framebuffer_y == 0 && $framebuffer_x == 0 ))`
although the arithmetic quecks print errors if values are no integers, thus for such simple checks I prefer string checks in most cases.